Thanks, you were right to question the `std::unordered_map` fallback.

I opened a RoboStack kilted probe PR to force the Windows rebuild with the downstream fallback removed: https://github.com/RoboStack/ros-kilted/pull/92

That reproduced the actual failure. The other platforms passed, and Windows failed while compiling `a_star.cpp` with MSVC errors in the vendored `robin_hood.h`:

```text
robin_hood.h(151,17): error C2059: syntax error: 'return'
robin_hood.h(152,67): error C2059: syntax error: 'constant'
robin_hood.h(153,13): error C2059: syntax error: '}'
robin_hood.h(153,13): error C2143: syntax error: missing ';' before '}'
```

The cause is the `// NOLINT` comment after a macro-continuation backslash in `ROBIN_HOOD_COUNT_TRAILING_ZEROES`, which prevents MSVC from continuing the macro body. I amended this PR to fix that macro line instead, so Windows keeps using `robin_hood::unordered_node_map` and avoids the performance concern.

--- a/nav2_smac_planner/include/nav2_smac_planner/thirdparty/robin_hood.h
+++ b/nav2_smac_planner/include/nav2_smac_planner/thirdparty/robin_hood.h
@@ -147,7 +147,8 @@ static Counts& counts() {
 #        pragma intrinsic(ROBIN_HOOD(BITSCANFORWARD))
 #        define ROBIN_HOOD_COUNT_TRAILING_ZEROES(x)                                       \
             [](size_t mask) noexcept -> int {                                             \
-                unsigned long index;                                                      \  // NOLINT
+                /* NOLINTNEXTLINE(runtime/int) */                                        \
+                unsigned long index;                                                      \
                 return ROBIN_HOOD(BITSCANFORWARD)(&index, mask) ? static_cast<int>(index) \
                                                                 : ROBIN_HOOD(BITNESS);    \
             }(x)

